### DISCUSSION TITLE
How to obtain the real user IP address.

### DISCUSSION THREAD

**Discussion by WuKaiYi** [tiangolo/uwsgi-nginx-flask-docker#383] [Questions] [answered]
client_ip = request.remote_addr

Using request.remote_addr to obtain the client's IP address always returns 172.21.0.1.

How can I obtain the real IP address?

**Comment by tiangolo** [comment, marked as answer]
You would probably need whatever is the termination proxy (maybe something like Traefik) to tell you what's the final user's IP in a header.

**Comment by omidboNBasT** [comment]
delaram@
